edge-http won't compile with `embassy-futures v0.1.2`

`embassy-futures v0.1.1 -> v0.1.2` is breaking due to https://github.com/embassy-rs/embassy/pull/3328 which fixed a soundness issue, which affects the inner workings of `edge-http`.

Compiling with `embassy-futures@v0.1.2` will produce the following compilation error:

```
error[E0308]: mismatched types
   --> ~/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/edge-http-0.6.0/src/io/server.rs:749:65
    |
718 |                 .push(async move {
    |                       ---------- the found `async` block
...
749 |         let (result, _) = embassy_futures::select::select_slice(&mut tasks).await;
    |                           ------------------------------------- ^^^^^^^^^^ expected `Pin<&mut [_]>`, found `&mut Vec<..., P>`
    |                           |
    |                           arguments to this function are incorrect
    |
    = note:         expected struct `Pin<&mut [_]>`
            found mutable reference `&mut Vec<{async block@~/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/edge-http-0.6.0/src/io/serve
r.rs:718:23: 718:33}, P>`
note: function defined here
   --> ~/.cargo/registry/src/index.crates.io-1949cf8c6b5b557f/embassy-futures-0.1.2/src/select.rs:523:8
    |
523 | pub fn select_slice<'a, Fut: Future>(slice: Pin<&'a mut [Fut]>) -> SelectSlice<'a, Fut> {
    |        ^^^^^^^^^^^^


```

--- a/Cargo.toml
+++ b/Cargo.toml
@@ -108,7 +108,7 @@ members = [
 ]
 
 [workspace.dependencies]
-embassy-futures = { version = "0.1", default-features = false }
+embassy-futures = { version = "0.1.2", default-features = false }
 embassy-sync = { version = "0.7", default-features = false }
 embassy-time = { version = "0.4", default-features = false }
 embedded-io-async = { version = "0.6", default-features = false }

--- a/edge-dhcp/CHANGELOG.md
+++ b/edge-dhcp/CHANGELOG.md
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+* Updated embassy-futures to 0.1.2
+
 ## [0.6.0] - 2025-05-29
 * Optional `defmt` support via two new features (one has to specify one, or the other, or neither, but not both):
   * `log` - uses the `log` crate for all logging
@@ -23,4 +27,3 @@ and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0
 * Migrated the client and the server to the `edge-nal` traits
 * Migrated the server to only require `UdpSend` and `UdpReceive`, without the need to manipulate raw IP payloads anymore
 * Raised MSRV to 1.77
-

--- a/edge-http/CHANGELOG.md
+++ b/edge-http/CHANGELOG.md
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+* Updated embassy-futures to 0.1.2
+
 ## [0.6.0] - 2025-05-29
 * Optional `defmt` support via two new features (one has to specify one, or the other, or neither, but not both):
   * `log` - uses the `log` crate for all logging

--- a/edge-http/src/io/server.rs
+++ b/edge-http/src/io/server.rs
@@ -746,7 +746,10 @@ impl<const P: usize, const B: usize, const N: usize> Server<P, B, N> {
                 .map_err(|_| ()));
         }
 
-        let (result, _) = embassy_futures::select::select_slice(&mut tasks).await;
+        let tasks = pin!(tasks);
+
+        let tasks = unsafe { tasks.map_unchecked_mut(|t| t.as_mut_slice()) };
+        let (result, _) = embassy_futures::select::select_slice(tasks).await;
 
         warn!(
             "Server processing loop quit abruptly: {:?}",

--- a/edge-mdns/CHANGELOG.md
+++ b/edge-mdns/CHANGELOG.md
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+* Updated embassy-futures to 0.1.2
+
 ## [0.6.1] - 2025-08-14
 * Fix HostAnswersMdnsHandler responding to any A/AAAA queries even without match
 

--- a/edge-nal-embassy/CHANGELOG.md
+++ b/edge-nal-embassy/CHANGELOG.md
@@ -5,6 +5,10 @@ All notable changes to this project will be documented in this file.
 The format is based on [Keep a Changelog](https://keepachangelog.com/en/1.0.0/),
 and this project adheres to [Semantic Versioning](https://semver.org/spec/v2.0.0.html).
 
+## Unreleased
+
+* Updated embassy-futures to 0.1.2
+
 ## [0.6.0] - 2025-05-29
 * Optional `defmt` support via two new features (one has to specify one, or the other, or neither, but not both):
   * `log` - uses the `log` crate for all logging

